src/R2_alg.jl

```diff
@@ -1,5 +1,56 @@
 export R2
 
+mutable struct R2Solver{R, S <: AbstractVector{R}} <: AbstractOptimizationSolver
+  xk::S
+  ∇fk::S
+  mν∇fk::S
+  xkn::S
+  s::S
+  has_bnds::Bool
+  l_bound::S
+  u_bound::S
+  l_bound_m_x::S
+  u_bound_m_x::S
+  Fobj_hist::Vector{Float64}
```

Comment:
It would be great to also parameterize the element type here, so we can use other types than Float64.